.ci/scripts/coverity.sh

```diff
@@ -0,0 +1,265 @@
+make_opt="-j$(($nproc / 2 + 1))"
+COV_BUILD_DIR=$(dirname $0)/cov-build
+
+# Path to known error filters file (should be part of repository)
+COV_ERROR_FILTERS_FILE=$(dirname $0)/errors.filters
+
+# Result file with all reported errors
+COV_RESULT_ALL_ERRORS_FILE=$(dirname $0)/errors.all
+
+# Result file with unfiltered errors
+COV_RESULT_UNFILT_ERRORS_FILE=$(dirname $0)/errors.unfiltered
+
+COV_ANALYSE_OPTIONS+=" --all"
+COV_ANALYSE_OPTIONS+=" --enable-fnptr"
+COV_ANALYSE_OPTIONS+=" --fnptr-models"
+COV_ANALYSE_OPTIONS+=" --checker-option INFINITE_LOOP:report_bound_type_mismatch:true"
+COV_ANALYSE_OPTIONS+=" --checker-option RESOURCE_LEAK:allow_unimpl:true"
+
+function show_usage() {
+        echo ""
+        echo ""
+        echo "Usage $(basename $0) [options]"
+        echo ""
+        echo "Run static code analysis using coverity, generate defects report and compare"
+        echo "current report with known issues filter file."
+        echo ""
+        echo "Script should run from main directory, after makefiles were generated."
+        echo ""
+        echo ""
+        echo "options:"
+        echo "--cov-build-args   <args>  Additional arguments for cov-build."
+        echo "                           For more information, see cov-build manual."
+        echo "--cov-analyze-args <args>  Additional arguments for cov-analyze."
+        echo "                           For more information, see cov-analyze manual."
+        echo ""
+        echo "--skip-coverity            Skip coverity stage."
+        echo "--skip-cov-build           Skip cov-build in coverity stage."
+        echo "--skip-filter-check        Skip Checking that each line in filters file (errors.filters),"
+        echo "                           filters out at least one issue."
+        echo ""
+        echo ""
+        echo "-h | --help                Show usage."
+        echo ""
+        echo ""
+}
+
+# Find and load Coverity
+function load_coverity() {
+        echo "Loading Coverity..."
+
+        local cov_build_path
+        cov_build_path='which cov-build'
+        if [ $? == 0 ]; then
+                local coverity_path=$(dirname ${cov_build_path})
+                echo "coverity already loaded. path: ${coverity_path}"
+                echo ""
+                return 0
+        fi
+
+        # Load Coverity module and verify that it was loaded successfully
+        module load ${COVERITY_MODULE_NAME} && module list 2>&1 | grep ${COVERITY_MODULE_NAME} > /dev/null
+        if [ $? == 0 ]; then
+                echo "coverity loaded from module ${COVERITY_MODULE_NAME}"
+                echo ""
+                return 0
+        fi
+
+        if [ -d ${DEFAULT_COVERITY_PATH} ]; then
+                PATH=${DEFAULT_COVERITY_PATH}:$PATH
+                echo "coverity loaded from path: ${DEFAULT_COVERITY_PATH}"
+                echo ""
+                return 0
+        fi
+
+        which cov-build
+        if [ $? != 0 ]; then
```

Comment:
Can shorten to
`if ! which cov-build > /dev/null; then`

Reply:
refactored and removed